messaging/src/main/java/org/axonframework/tracing/NestingSpanFactory.java

```diff
@@ -0,0 +1,176 @@
+     * Creates the {@link NestingSpanFactory} based on the {@link Builder} provided.
+     * @param builder The {@link Builder} to use during construction.
+     */
+    protected NestingSpanFactory(Builder builder) {
+        this.delegateSpanFactory = builder.delegateSpanFactory;
+        this.timeLimit = builder.timeLimit;
+        this.clock = builder.clock;
+    }
+
+    /**
+     * Creates a new {@link Builder} that can build a {@link NestingSpanFactory}.
+     * @return The {@link Builder} in charge of creating a {@link NestingSpanFactory}.
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    @Override
+    public Span createRootTrace(Supplier<String> operationNameSupplier) {
+        return delegateSpanFactory.createRootTrace(operationNameSupplier);
+    }
+
+    @Override
+    public Span createHandlerSpan(Supplier<String> operationNameSupplier, Message<?> parentMessage,
+                                  boolean isChildTrace, Message<?>... linkedParents) {
+        if (!isChildTrace && messageIsWithinTimeLimit(parentMessage)) {
+            return delegateSpanFactory.createHandlerSpan(operationNameSupplier, parentMessage, true, linkedParents);
+        }
+        return delegateSpanFactory.createHandlerSpan(operationNameSupplier, parentMessage, isChildTrace, linkedParents);
+    }
+
+    private boolean messageIsWithinTimeLimit(Message<?> parentMessage) {
+        if (!(parentMessage instanceof EventMessage)) {
+            return true;
+        }
+        Instant timestamp = ((EventMessage<?>) parentMessage).getTimestamp();
+        return clock.instant().isBefore(timestamp.plus(timeLimit));
+    }
+
+    @Override
+    public Span createDispatchSpan(Supplier<String> operationNameSupplier, Message<?> parentMessage,
+                                   Message<?>... linkedSiblings) {
+        return delegateSpanFactory.createDispatchSpan(operationNameSupplier, parentMessage, linkedSiblings);
+    }
+
+    @Override
+    public Span createInternalSpan(Supplier<String> operationNameSupplier) {
+        return delegateSpanFactory.createInternalSpan(operationNameSupplier);
+    }
+
+    @Override
+    public Span createInternalSpan(Supplier<String> operationNameSupplier, Message<?> message) {
+        return delegateSpanFactory.createInternalSpan(operationNameSupplier, message);
+    }
+
+    @Override
+    public void registerSpanAttributeProvider(SpanAttributesProvider provider) {
+        delegateSpanFactory.registerSpanAttributeProvider(provider);
+    }
+
+    @Override
+    public <M extends Message<?>> M propagateContext(M message) {
+        return delegateSpanFactory.propagateContext(message);
+    }
+
+
+    /**
+     * Creates a builder that will create a {@link NestingSpanFactory}.
+     * <p>
+     * Requires the delegate {@link SpanFactory} to be configured.
+     * <p>
+     * The {@link Clock} defaults to the system utc time and the timeLimit {@link Duration} defaults to two minutes.
```

Comment:
I think it is good if we add the defaults to the `builder()` method too. Chances are high that users *do* read the method's javadoc, compared to the class' javadoc.